### PR TITLE
new: bsm powermeter driver

### PR DESCRIPTION
**Pull request by chrkli22**
- implements powermeter interface, including "get_signed_meter_value", which returns a signed OCMF snapshot string.
- publishes variable "powermeter" of type Powermeter.
- type Powermeter: added optional VAR property.
- cmd for requesting standard sunspec ac meter model.
- uses SerialCommHub or direct modbus.

Module currently runs without problems on its testinstallation.

### CODE REVIEW

File: modules/PowermeterBSM/lib/known_model.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+/*
+ * Warning: only specify *addresses* relative to the sunspec base address,
+ * the transport layer adds the relative addresses to the sunspec base address when querying the device.
+ */
+
+const AddressData Sunspec_Common{3_sma, calc_model_size_in_register(sunspec_model::Common::Model)};
+const AddressData Sunspec_ACMeter{91_sma, calc_model_size_in_register(sunspec_model::ACMeter::Model)};
+
+// TODO: this starts reading two registers after the "official beginning of the model, we dont need model id and payload
```
Comment (SebaLukas) [inline modules/PowermeterBSM/lib/known_model.cpp:16]: Is that solved?

File: modules/PowermeterBSM/lib/transport.cpp [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+        static_assert(sizeof(swapit.val_32) == sizeof(swapit.val_8));
+
+        transport::DataVector tmp{};
+
+        for (auto item : serial_com_hub_result.value.get()) {
+            swapit.val_32 = item;
+            tmp.push_back(swapit.val_8.v2);
+            tmp.push_back(swapit.val_8.v3);
+            // EVLOG_debug << __PRETTY_FUNCTION__ << " " << std::hex << item << " v0: " << (int)swapit.val_8.v0 << " v1
```
Comment (SebaLukas) [inline modules/PowermeterBSM/lib/transport.cpp:135]: Delete comments

File: modules/PowermeterBSM/lib/transport.hpp [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * data transport via modbus.
+ */
+class ModbusTransport : public AbstractDataTransport {
+
+protected:
+    everest::connection::SerialDeviceConfiguration m_cfg;
+    everest::connection::SerialDevice m_device;
+    // everest::connection::SerialDeviceLogToStream      m_device;
```
Comment (SebaLukas) [inline modules/PowermeterBSM/lib/transport.hpp:61]: Delete or keep?

File: modules/PowermeterBSM/main/powermeterImpl.cpp [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+                transport::AbstractDataTransport::Spt transport = mod->get_data_transport();
+
+                EVLOG_debug << __PRETTY_FUNCTION__ << " wakeup. " << std::endl;
+
+                transport::DataVector data = transport->fetch(known_model::Sunspec_ACMeter);
+
+                sunspec_model::ACMeter acmeter(data);
+                types::powermeter::Powermeter result;
+
+                result.timestamp = 0; // FIXME: current implementation is a float that cant store a 32/31 bit
```
Comment (SebaLukas) [inline modules/PowermeterBSM/main/powermeterImpl.cpp:108]: Is that solved?

File: modules/PowermeterBSM/manifest.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
+{
+	"description": "Module that collects power and energy measurements from a MODBUS RTU device",
```
Comment (SebaLukas) [inline modules/PowermeterBSM/manifest.json:2]: Delete JSON file

**Review by SebaLukas — CHANGES_REQUESTED** [review]
Looks nice. Can you look into the changes i request. Then merging is no problem :D

File: modules/PowermeterBSM/lib/known_model.cpp (hunk 1 above)
Comment (chrkli22) [reply modules/PowermeterBSM/lib/known_model.cpp:16]: No, this is just a reminder that its starts reading a bit later. Does not make any difference: i will remove the TODO

File: modules/PowermeterBSM/lib/transport.hpp (hunk 3 above)
Comment (chrkli22) [reply modules/PowermeterBSM/lib/transport.hpp:61]: i would like to keep this as a reminder that the implementation with a logstream exists. This is usesful for debugging.

File: modules/PowermeterBSM/main/powermeterImpl.cpp (hunk 4 above)
Comment (chrkli22) [reply modules/PowermeterBSM/main/powermeterImpl.cpp:108]: No, this is a result of the code generated by ev-cli that renders an interface definition "number" as float. This is kept for backward compatibility.

File: modules/PowermeterBSM/manifest.json (hunk 5 above)
Comment (chrkli22) [reply modules/PowermeterBSM/manifest.json:2]: fixed

**Review by SebaLukas — APPROVED** [review]

**Pull request merged**